pxtblocks/blocklyloader.ts

```diff
@@ -750,6 +760,48 @@ namespace pxt.blocks {
         });
     }
 
+    /**
+     * The following patch to blockly is to add the Trash icon on top of the toolbox, 
+     * the trash icon should only show when a user drags a block that is already in the workspace.
+     */
+    function initDrag() {
+        const calculateDistance = (elem: any, mouseX: any) => {
+            return Math.floor(mouseX - (elem.offset().left + (elem.width() / 2)));
+        }
+        /**
+         * Track a drag of an object on this workspace.
+         * @param {!Event} e Mouse move event.
+         * @return {!goog.math.Coordinate} New location of object.
+         */
+        let moveDrag = (<any>Blockly).WorkspaceSvg.prototype.moveDrag;
+        (<any>Blockly).WorkspaceSvg.prototype.moveDrag = function(e: any) {
```

Comment:
add moveDrag to blockly.d.ts so that we keep track of internal functions being used